File: src/main/java/io/myosotisdev/utopianism/modules/stat/ArrowParticles.java
package io.myosotisdev.utopianism.modules.stat;

import io.myosotisdev.utopianism.modules.effect.ParticleEffect;
import io.myosotisdev.utopianism.modules.stat.data.ItemStatData;
import io.myosotisdev.utopianism.modules.stat.data.ParticleEffectData;
import io.myosotisdev.utopianism.serialize.ParticleTagSerializer;
import io.myosotisdev.utopianism.serialize.Serializer;
import io.myosotisdev.utopianism.util.Tags;
import net.minestom.server.item.ItemStack;
import net.minestom.server.tag.Tag;

public class ArrowParticles
        extends ItemStat<ParticleEffect>
{


    ArrowParticles()
    {
        super("arrow-particles", Tags.createTag("arrow-particles", ParticleEffectData.Serializer));


    }

    @Override
    public void onApply(ItemStack itemStack, ItemStatData data)
    {

    }

    @Override
    public ItemStatData defaultStatData()
    {
        return new ParticleEffectData();
    }
}

File: src/main/java/io/myosotisdev/utopianism/modules/stat/ArrowParticles.kt
package io.myosotisdev.utopianism.modules.stat

import io.myosotisdev.utopianism.modules.effect.ParticleEffect
import io.myosotisdev.utopianism.modules.stat.data.ItemStatData
import io.myosotisdev.utopianism.modules.stat.data.ParticleEffectData
import io.myosotisdev.utopianism.util.Tags
import net.minestom.server.item.ItemStack
import net.minestom.server.tag.Tag

class ArrowParticles() : ItemStat<ParticleEffect>("arrow-particles", Tags.createTag("arrow-particles", ParticleEffectData.Serializer) as Tag<ParticleEffect>)
{
    override fun onApply(itemStack: ItemStack, data: ItemStatData<in ParticleEffect>)
    {

    }

    override fun defaultStatData(): ItemStatData<ParticleEffect>?
    {
        return ParticleEffectData()
    }
}